pkg/castor/client.go

```diff
@@ -0,0 +1,66 @@
+//
+// Copyright (c) 2022 - for information on the respective copyright owner
+// see the NOTICE file and/or the repository https://github.com/carbynestack/ephemeral.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+package castor
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/asaskevich/govalidator"
+)
+
+// AbstractClient is an interface for castor tuple client.
+type AbstractClient interface {
+	GetTuples(int32, TupleType, string) (TupleList, error)
+}
+
+func NewCastorClient(u url.URL) (*Client, error) {
+	ok := govalidator.IsURL(u.String())
+	if !ok {
+		return &Client{}, errors.New("invalid Url")
+	}
+	httpClient := http.Client{}
+	return &Client{HttpClient: httpClient, Url: u}, nil
+}
+
+type Client struct {
+	Url        url.URL
+	HttpClient http.Client
```

Comment:
the client is used internally only and not passed around. Access is in complete control of the client. 
However, it could also be changed to a pointer for convenience.